### PR TITLE
fix bug: pathMap cache need to delete after node removed

### PR DESCRIPTION
**Pull request by cadeeper**
as the title mentioned.

### CODE REVIEW

**Review by zouyx — COMMENTED** [review]
What will happen if it is not deleted?

**Comment by cadeeper** [comment]
> What will happen if it is not deleted?

node add event won't trigger when the same provider node unregister and regiester again. it will continue on : @zkPath %s has already been listened.

File: remoting/zookeeper/listener.go [hunk 1]
```diff
@@ -301,6 +301,9 @@ func (l *ZkEventListener) listenDirEvent(conf *common.URL, zkPath string, listen
 			go func(zkPath string, listener remoting.DataListener) {
 				if l.listenServiceNodeEvent(zkPath) {
 					listener.DataChange(remoting.Event{Path: zkPath, Action: remoting.EventTypeDel})
+					l.pathMapLock.Lock()
+					delete(l.pathMap, zkPath)
+					l.pathMapLock.Unlock()
```
Comment (AlexStocks) [inline remoting/zookeeper/listener.go:306]: plg using defer here.

Comment (cadeeper) [reply remoting/zookeeper/listener.go]: done

**Review by AlexStocks — APPROVED** [review]

**Review by gaoxinge — APPROVED** [review]

**Review by cityiron — APPROVED** [review]

**Review by fangyincheng — APPROVED** [review]

**Pull request merged**